### PR TITLE
User/chgray/clog 0.3.0 staging

### PR DESCRIPTION
**Pull request by chgray**
draft folding in clog 0.3.0

### CODE REVIEW

**Review by nibanks — COMMENTED** [review]
You also seem to have some empty install powershell script.

**Comment by ThadHouse** [comment]
One other thing I realized with this PR is we never enable clog on macOS. However, with the existing builds that does work, but I know based on the clog side it will not. So something I'd want to test and ensure is working before merging.

File: src/core/binding.c [hunk 1]
```diff
@@ -158,8 +158,8 @@ QuicBindingInitialize(
         "[bind][%p] Created, Udp=%p LocalAddr=%!ADDR! RemoteAddr=%!ADDR!",
         Binding,
         Binding->Socket,
-        CLOG_BYTEARRAY(sizeof(DatapathLocalAddr), &DatapathLocalAddr),
-        CLOG_BYTEARRAY(sizeof(DatapathRemoteAddr), &DatapathRemoteAddr));
+        CLOG_BYTEARRAY_BUGBUG(sizeof(DatapathLocalAddr), &DatapathLocalAddr),
+        CLOG_BYTEARRAY_BUGBUG(sizeof(DatapathRemoteAddr), &DatapathRemoteAddr));
```
[the remark this answers is not in the document]
Comment (chgray) [reply src/core/binding.c:162]: disagree.  type safety matters and relying on casting oddities is bad news - i'd be okay with you writing some kind of cast macro "ADDR_TO_PTR"?

File: src/core/connection.c [hunk 2]
```diff
@@ -789,7 +789,7 @@ QuicConnUpdateRtt(
     if (RttUpdated) {
         CXPLAT_DBG_ASSERT(Path->SmoothedRtt != 0);
         QuicTraceLogConnVerbose(
-            RttUpdated,
+            RttUpdatedMsg,
```
[the remark this answers is not in the document]
Comment (chgray) [reply src/core/connection.c:792]: manifested ETW collides here - it generates a macro

**Comment by ThadHouse** [comment]
The docker windows build should not be a part of this PR. This PR is already too big as it is. We don't need to make it more complicated. That should be done in a separate PR.

**Comment by nibanks** [comment]
> The docker windows build should not be a part of this PR. This PR is already too big as it is. We don't need to make it more complicated. That should be done in a separate PR.

Big +1. This PR already needed to be broken up to more manageable pieces.

File: src/manifest/MsQuicEtw.man [hunk 3]
```diff
@@ -3012,905 +10059,3493 @@
...
+            id="CLOG.IndicateConnectionShutdownComplete"
+            value="[conn][%0 ] Indicating QUIC_CONNECTION_EVENT_SHUTDOWN_COMPLETE"
+            />
+        <string
+            id="CLOG.IndicateResumptionTicketReceived"
+            value="[conn][%0 ] Indicating QUIC_CONNECTION_EVENT_RESUMPTION_TICKET_RECEIVED"
+            />
+        <string
+            id="CLOG.ClientVersionNegotiationCompatibleVersionUpgrade"
+            value="[conn][%0 ] Compatible version upgrade! Old: 0x%1 , New: 0x%2 "
```
Comment (nibanks) [inline src/manifest/MsQuicEtw.man:11268]: I would fix this if I knew how to in CLOG, but it seems that CLOG is adding extra spaces after all the `%N`s in the strings.

**Pull request closed without merging**